platform-operator/metricsexporter/metricsexporter_internal.go

```
@@ -0,0 +1,185 @@
+		"opensearch":                      opensearchMetrics,
+		"opensearch-dashboards":           opensearchDashBoardsMetrics,
+		"grafana":                         grafanaMetrics,
+		"coherence-operator":              coherenceMetrics,
+		"mysql":                           mySQLMetrics,
+		"keycloak":                        keycloakMetrics,
+		"kiali-server":                    kialiMetrics,
+		"prometheus-operator":             prometheusOperatorMetrics,
+		"prometheus-adapter":              prometheusAdapterMetrics,
+		"kube-state-metrics":              kubeStateMetricsMetrics,
+		"prometheus-pushgateway":          prometheusPushGatewayMetrics,
+		"prometheus-node-exporter":        prometheusNodeExporterMetrics,
+		"jaeger-operator":                 jaegerOperatorMetrics,
+		"verrazzano-console":              verrazzanoConsoleMetrics,
+		"fluentd":                         fluentdMetrics,
+	}
+
+	// This array will be populated with all the metrics from each map
+	// Metrics not included in a map can be added to this array for registration
+	allMetrics = populateAllMetricsList(metricsMap)
+	// This map will be automatically populated with all metrics which were not registered correctly
+	// Metrics in this map will be retried periodically
+	failedMetrics = map[prometheus.Collector]int{}
+
+	registry = prometheus.DefaultRegisterer
+)
+
+// This function is used to determine whether a durationTime for a component metric should be set and what the duration time is
+// If the start time is greater than the completion time, the metric will not be set
+// After this check, the function calculates the duration time and tries to set the metric of the component
+// If the component's name is not in the metric map, an error will be raised to prevent a seg fault
+func metricParserHelperFunction(log vzlog.VerrazzanoLogger, componentName string, startTime string, completionTime string, typeofOperation string) {
+	startInSeconds, _ := time.Parse(time.RFC3339, startTime)
```

Why are we ignoring the errors for `time.Parse`?

---

it's generally best to not make such assumptions. What happens if this is called with a value that for some reason doesn't parse (maybe we cannot think of the reason until it happens)

---

Thanks!